crypto/des/qud_cksm.c

```diff
@@ -65,10 +68,9 @@ DES_LONG DES_quad_cksum(const unsigned char *input, DES_cblock output[],
             t0 &= 0xffffffffL;
             t1 = z1;
             /* square, well sort of square */
-            z0 = ((((t0 * t0) & 0xffffffffL) + ((t1 * t1) & 0xffffffffL))
-                  & 0xffffffffL) % 0x7fffffffL;
-            z1 = ((t0 * ((t1 + NOISE) & 0xffffffffL)) & 0xffffffffL) %
-                0x7fffffffL;
+            z0 = ((((uint64_t)t0 * t0) & 0xffffffffL) +  
+                  (((uint64_t)t1 * t1) & 0xffffffffL)) % 0x7fffffffL;
+            z1 = (((uint64_t)t0 * ((t1 + NOISE) & 0xffffffffL)) & 0xffffffffL) %
```

Comment:
These changes are unnecessary due to the masking with `0xffffffffL`. `DES_LONG` is unsigned so overflow is well-defined, not UB.

Please, when using static code analysis tools, read and understand the code before suggesting changes.